### ISSUE TITLE
Foundations API does not return complete information for commercial data

### ISSUE THREAD

**Issue by david-silverink** [reapit/foundations#5282]
Payload is missing type, situation and style fields when returned, which is available for residential properties.
This data is required in order to fully facilitate search and filtering tools for customers.

**Comment by plittlewood-rpt** [comment]
Hi @david-silverink / @awogan06 

We need to do some research about how the wider customer base is using the Commercial mechanisms in Agency Cloud before we can decide how best to implement this support as it doesn't really fit the current contract. Options used for residential stock are very similar between customers, but that is not the case for commercial properties so the problem is more complex.

I have logged a ticket to cover the research exercise #5350 from which we'll be able to make more informed decisions about the route to take.

Thanks
Pete

**Comment by david-silverink** [comment]
Hi @plittlewood-rpt We are experiencing a similar issue but for a residential property (NWR170333) and I wonder if you could shed further light on why this particular property's type (residential land/site) would not feed, but another, e.g. CRK210543 seems to work fine type 'house'

Thanks Pete

@awogan06

**Comment by plittlewood-rpt** [comment]
Hi @david-silverink - this is because we have had to come up with a standard approach to how property attributes are exposed in the Platform APIs to ensure that developers can build apps that work with any system. To do this we did a whole tranch of research to come up with the commonly used attributes across our customer base which allowed us to come up with a standard 'set' and any appropriate variations (ie "land" has variations of "Land" and "Land Site"). We are actually literally about to merge in some changes to add some additional variations to some of the existing attributes and I may be able to get this in as part of #5235 - "Residential Site/Land" will be added as a mapping to the existing option `land` so the property you refer to above will be exposed in the Platform API as such. Does this work for you?

**Comment by david-silverink** [comment]
That would be great Pete, but will any of the other types used by SFG be similarly affected and rely on a further update? We want to be confident that all residential property types selected in AC will be exposed to the API (bar commercial types, which I understand to be implemented separately).

**Comment by david-silverink** [comment]
@plittlewood-rpt

**Comment by david-silverink** [comment]
Thanks @plittlewood-rpt. Could I please check that this body of work will allow the SOAP API to be omitted from our workflow completely? I am conscious that there are one or two other gaps in functionality that may or may not be accounted for through the implementation of the commercial attributes in the new API. For example, the SFG custom residential type 'Residential Land/Site' does not exist in the new API, even though it is not a commercial type. We currently need to use the SOAP API for all attributes as a result, to be sure to capture everything. @awogan06

**Comment by david-silverink** [comment]
Thanks @plittlewood-rpt , however I think it’s more that we need some assurance that any attributes be they custom or commercial are accessible in the new API and that we can stop using the SOAP endpoints. Currently we can access everything through SOAP.

**Comment by plittlewood-rpt** [comment]
Hi @david-silverink - just to update you on this, we are nearing completion of some preliminary work that will allow us to more easily support the commercial attributes. I'm hopeful this will go out this week after which we can start properly planning the commercial attribute support, so you should see movement on this ticket in the next week or so.

**Comment by david-silverink** [comment]
Thats great, thanks @plittlewood-rpt

**Comment by plittlewood-rpt** [comment]
I have started to break this out in separate work items in [this milestone](https://github.com/reapit/foundations/milestone/129)

**Comment by HollyJoyPhillips** [comment]
@david-silverink We have completed some development work which deals with some of these items, please can you review and let us know your thoughts? Thanks.

**Comment by HollyJoyPhillips** [comment]
@david-silverink We are closing this ticket as we haven't had a response. If there are any outstanding data fields that are still required, please open a new ticket for us to review.
